pkg/secretutils/secret_manager.go

```diff
@@ -125,16 +138,54 @@ func (sm *SecretManager) obtainSecretForOwner(key types.NamespacedName, owner cl
 // ensure it is present in the cache (and that we can watch for changes), and
 // that it has a particular owner reference. The owner reference may optionally
 // be a controller reference.
-func (sm *SecretManager) AcquireSecret(key types.NamespacedName, owner client.Object, ownerIsController bool) (*corev1.Secret, error) {
+func (sm *SecretManager) AcquireSecret(key types.NamespacedName, owner client.Object, ownerIsController, addFinalizer bool) (*corev1.Secret, error) {
 	if owner == nil {
 		panic("AcquireSecret called with no owner")
 	}
 
-	return sm.obtainSecretForOwner(key, owner, ownerIsController)
+	return sm.obtainSecretForOwner(key, owner, ownerIsController, addFinalizer)
 }
 
 // ObtainSecret retrieves a Secret and ensures that it has a label that will
 // ensure it is present in the cache (and that we can watch for changes).
 func (sm *SecretManager) ObtainSecret(key types.NamespacedName) (*corev1.Secret, error) {
-	return sm.obtainSecretForOwner(key, nil, false)
+	return sm.obtainSecretForOwner(key, nil, false, false)
+}
+
+// ReleaseSecret removes secrets manager finalizer from specified secret when needed.
+func (sm *SecretManager) ReleaseSecret(secret *corev1.Secret) error {
+	if !utils.StringInList(secret.Finalizers, SecretsFinalizer) {
+		return nil
+	}
+
+	// Remove finalizer from secret to allow deletion
+	secret.Finalizers = utils.FilterStringFromList(
+		secret.Finalizers, SecretsFinalizer)
+
+	if err := sm.client.Update(context.Background(), secret); err != nil {
+		return errors.Wrap(err, fmt.Sprintf("failed to remove finalizer from secret %s in namespace %s",
+			secret.ObjectMeta.Name, secret.ObjectMeta.Namespace))
+	}
+
+	sm.log.Info("removed secret finalizer",
+		"remaining", secret.Finalizers)
+
+	return nil
+}
+
+// ReleaseSecretByRef removes secrets manager finalizer from secret specified be reference when needed.
+func (sm *SecretManager) ReleaseSecretByRef(secref *corev1.SecretReference) error {
```

Comment:
This is not actually used any more, is it? Let's just remove it.

Reply:
not used, yes, was thinking to keep it for consistent public interface for SecretManager, just in case we start to think about other secrets with finalizers, will remove